TheMET/TheMET/FullScreenPhotoViewController/FullScreenPhotoViewController.swift

```diff
@@ -47,4 +75,78 @@ class FullScreenPhotoViewController: UIViewController {
         self.dismiss(animated: true)
     }
     
+    @objc
+    private func handlePinchGesture(_ recognizer: UIPinchGestureRecognizer) {
+        switch recognizer.state {
+        case .began:
+            self.currentGestureScale = self.scale
+        case .ended:
+            break
+        case .changed:
+            let scale = self.currentGestureScale * recognizer.scale
+            self.applyTransform(transX: self.transX, transY: self.transY, scale: scale)
+        case .failed, .cancelled:
+            self.scale = self.currentGestureScale
+            self.tranformImageView()
+        default: break
+        }
+    }
+    
+    @objc
+    private func handlePanGesture(_ recognizer: UIPanGestureRecognizer) {
+        switch recognizer.state {
+        case .began:
+            self.currentGestureTransX = self.transX
+            self.currentGestureTransY = self.transY
+        case .ended: break
+        case .changed:
+            let tranform = recognizer.translation(in: self.view)
+            let transX = self.currentGestureTransX + tranform.x/self.scale
+            let transY = self.currentGestureTransY + tranform.y/self.scale
+            self.applyTransform(transX: transX, transY: transY, scale: self.scale)
+        case .failed, .cancelled:
+            self.transX = self.currentGestureTransX
+            self.transY = self.currentGestureTransY
+            self.tranformImageView()
+        default: break
+        }
+    }
+    
+    private func applyTransform(transX: CGFloat, transY: CGFloat, scale: CGFloat) {
+        self.scale = CGFloat.value(original: scale, max: self.maxScale, min: self.minScale)
+        let imageSize = self.imageSize
+        let viewSize = self.viewSize
+        let permissibleTransX: CGFloat = self.calculatePermissibleTrans(viewSize: viewSize.width, imageSize: imageSize.width)
+        let permissibleTransY: CGFloat = self.calculatePermissibleTrans(viewSize: viewSize.height, imageSize: imageSize.height)
+        self.transX = CGFloat.value(original: transX, max: permissibleTransX, min: -permissibleTransX)
+        self.transY = CGFloat.value(original: transY, max: permissibleTransY, min: -permissibleTransY)
+        self.tranformImageView()
+    }
+    
+    private func calculatePermissibleTrans(viewSize: CGFloat, imageSize: CGFloat ) -> CGFloat {
+        let restRelWidth = 0.1 + (self.scale*(viewSize - imageSize)/2)/viewSize
```

Comment:
`0.1` is a `magic number here` please move it to another contents like `maxScale/minScale`